Gemfile

```diff
@@ -11,18 +11,19 @@ gem "elasticsearch-rails"
 gem "faraday"
 gem "faraday_middleware"
 gem "govuk_elements_rails"
-gem "govuk_publishing_components"
+gem "govuk_publishing_components", "~> 28.7.1" # Earliest release with Ruby 3+ support
```

Comment:
RE:

> The incompatibility was resolved in v31.1.0, but that requires a newer version of Puma that seems to be incompatible with the version of Capybara that this repo uses(?)

Can you provide more details on the incompatibility, e.g. a stack trace showing the failure it causes?

Also is there another version between 28.7.1 and 31.1.0 that still works? It looks as though you've chosen the oldest possible version that supports Ruby 3+, but we probably want to use the newest possible version that doesn't break anything.